src/Gaffer/TweakPlug.cpp

```diff
@@ -247,7 +247,17 @@ bool TweakPlug::applyTweak( IECore::CompoundData *parameters, MissingMode missin
 			{
 				return parameters->writable().erase( valueName ) > 0;
 			}
-			parameters->writable()[valueName] = newData;
+
+			CompoundDataMap &data = parameters->writable();
+			if( data[valueName] && data[valueName]->typeId() == InternedStringData::staticTypeId() )
+			{
+				if( const StringData *s = runTimeCast<const StringData>( newData.get() ) )
+				{
+					data[valueName] = new InternedStringData( s->readable() );
+					return true;
+				}
+			}
+			data[valueName] = newData;
```

Comment:
It's probably not important in the scheme of things, but it's bothering me a bit that we now have three map lookups where there used to be one. One way of avoiding it while keeping the same structure you've got might be this?

```
DataPtr &dst = parameters->writable()[valueName];
if( runTimeCast<InternedStringData>( dst.get() ) )
{
    if( ... )
    {
        dst = new InternedStringData( ... );
        return true;
    }
}
dst = newData;
```

Reply:
Fixed in 0c5591ca2a40b281ed6b7b2db82bd52d2d7c40d9